packages/hebao_v1/contracts/modules/base/BaseModule.sol

```diff
@@ -25,20 +25,14 @@ abstract contract BaseModule is Module
     event Activated   (address wallet);
     event Deactivated (address wallet);
 
-    struct ControllerCache
-    {
-        PriceOracle      priceOracle;
-        address          collectTo;
-    }
-
-    ControllerCache public controllerCache;
-
     ModuleRegistry public immutable moduleRegistry;
     SecurityStore  public immutable securityStore;
     WhitelistStore public immutable whitelistStore;
     QuotaStore     public immutable quotaStore;
     HashStore      public immutable hashStore;
     address        public immutable walletFactory;
+    PriceOracle    public immutable priceOracle;
+    address        public immutable collectTo;
```

Comment:
In ControllerImpl, there is a `setCollectTo` function, if collectTo is immutable in modules, then the modules won't be able to use any new value. We have to consider the scenario that we sometimes have to change the fee recipient address.

Reply:
But can't we just handle that in the `collectTo` contract? This can just be a simple contract that is able to set a dynamic recipient address. It's cheaper to make it configurable there than in the modules because in the modules we have to read it in for each transaction.

Reply:
I think these two solutions will end up using the same gas, the real collectTo address inside the collectTo contract will also have to be read before transferring tokens to it.

Reply:
I don't think the real `collectTo` needs to be read. The `collectTo` would just be a smart contract collecting all funds, so there's no need for the modules to know anything else. This smart contract at a fixed address collects all fees as they come in. Then if we actually want to send those fees to the real destination address it will be possible to specify a destination address (can even be passed in as a function argument).

The immutable `collectTo` address that's used in the modules can just be a simple `Drainable` contract. It's even possible to put the contract behind a proxy to make the contract itself updatable if that would be wanted.

Reply:
No need to make it upgradable or use a contract then. Could you please add a simple FeeCollector contract which is drainable by owner?